Allow user to change and update the personal information.

aynorway opened the pull request:
Account information here are not allowed to be changed or edited. Account information are Email and Username.
New user.html
New user_edit.html

aynorway committed:
Allow user to change and update the personal information.
Account information here are not allowed to be changed or edited.
Account information are Email and Username.
New user.html
New user_edit.html

aynorway committed:
Merge branch 'main' into feature/change_profile

aynorway committed:
add user.py

aynorway committed:
Merge remote-tracking branch 'origin/feature/change_profile' into feature/change_profile

aynorway committed:
prevent SQL injection

aynorway committed:
prevent SQL injection

singmnj commented on price-matching-system.pyproj:
```diff
@@ -28,6 +28,7 @@
     <EnableUnmanagedDebugging>false</EnableUnmanagedDebugging>
   </PropertyGroup>
   <ItemGroup>
+    <Compile Include="price_matching_system\auth - Copy.py" />
```
this seems like a leftover reference to an unused file.
should be deleted.

aynorway replied:
Thanks, a great one.

aynorway replied:
deleted

singmnj commented on price_matching_system/user.py:
```diff
@@ -0,0 +1,91 @@
+    conn = get_conn()
+    cursor = conn.cursor()
+    query = f"""SELECT * FROM dbo.tbl_User u
+                    WHERE UserId = {session.get('login_id')}"""
```
login id is not the same as user id
use user id from the g object here

aynorway replied:
userId is from the table and the POST data should be connected to the table.

aynorway replied:
fixed

singmnj commented on price_matching_system/user.py:
```diff
@@ -0,0 +1,91 @@
+    return render_template("user_edit.html", personalInfo=personalInfo)
+
+
+@bp.route("/userSave", methods=("GET", "POST"))
```
you can use the /user route with post verb here
no need of a new route name

aynorway replied:
Can be checked again

aynorway replied:
changed to post

singmnj commented on price_matching_system/templates/user.html:
```diff
@@ -0,0 +1,74 @@
+        <hr />
+        <div class="mx-auto mt-5 p-2">
+            <label class="form-label" for="email">Email</label>
+            <input id="email" class="form-control form-control-lg" value="{{ g.user['Email'] }}" readonly>
```
can we not use inputs in the display page?
it confuses the user whether they can edit or not

aynorway replied:
changed to <div>

singmnj commented on price_matching_system/templates/user.html:
```diff
@@ -0,0 +1,74 @@
+{% block content %}
+
+<div class="row mt-5">
+    <div class="col-md-4 mx-auto p-5" style="min-width: 500px; border: solid black">
```
can we use two columns to save vertical space ?

aynorway replied:
UI can be decided again for sure.

aynorway replied:
changed to two columns

singmnj commented on price_matching_system/templates/layout.html:
```diff
@@ -32,26 +32,28 @@
-                                <li><a class="dropdown-item" href="#">My Profile</a></li>
-                                <li><a class="dropdown-item" href="#">Change Details</a></li>
+                                <li><a class="dropdown-item" href="/user">My Profile</a></li>
+                                <li><a class="dropdown-item" href="/userEdit">Change Details</a></li>
```
can we name the route differently? like /user/edit

aynorway replied:
if for popular way, able to be changed, sure.

aynorway replied:
changed to /user/edit

singmnj commented on price_matching_system/user.py:
```diff
@@ -0,0 +1,91 @@
+    cursor.execute(query)
+    columns = [column[0] for column in cursor.description]
+    row = cursor.fetchone()
+    personalInfo = dict(zip(columns, row))
```
can we have empty string instead of None when the column value is null in the database ?

aynorway replied:
Not understand.

aynorway replied:
changed to ' '

singmnj commented on price_matching_system/user.py:
```diff
@@ -0,0 +1,91 @@
+
+@bp.route("/userEdit")
+def userEdit():
+    conn = get_conn()
```
this code is exactly same for the previous route
can we separate it out in a common method ?

aynorway replied:
yes sure, they are indeed copied.

aynorway replied:
Sure

singmnj commented on price_matching_system/user.py:
```diff
@@ -0,0 +1,91 @@
+
+    if error is None:
+        try:
+            query = f"UPDATE dbo.tbl_User SET [FirstName] = ? " \
```
again should be saving empty values as NULL in DB not as None

aynorway replied:
This is a method for preventing SQL injection threaten

aynorway replied:
Changed to ' '

singmnj reviewed (commented):
need changes

aynorway committed:
1.g.user['UserId'] fixed
2.user route with POST updated
3.User page <input> to <div>
4.route /user/edit
5.NULL/NONE to ' '

aynorway commented:
All feedback addressed, please have a look
